overlord/ifacestate/handlers.go

```diff
@@ -244,12 +252,20 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	var snapst snapstate.SnapState
 	plug := m.repo.Plug(plugRef.Snap, plugRef.Name)
+	if err := snapstate.Get(st, plugRef.Snap, &snapst); err != nil {
+		return err
+	}
+	if err := setupSnapSecurity(task, plug.Snap, snapst.DevMode(), m.repo); err != nil {
+		return err
+	}
+
 	slot := m.repo.Slot(slotRef.Snap, slotRef.Name)
-	if err := setupSnapSecurity(task, plug.Snap, m.repo); err != nil {
+	if err := snapstate.Get(st, slotRef.Snap, &snapst); err != nil {
```

Comment:
this is one of the reasons why the Get interface is not so great, I think to be on the safe side you need to use a 2nd SnapState variable here (there's a risk of picking up values left there from the first Get)